examples/parallel_profiling.py

```diff
@@ -0,0 +1,164 @@
+
+To illustrate those differences, the moving average from a large array will be
+computed. Therefore, the average can be computed in parallel at different
+location of the signal.
+
+"""
+
+###############################################################################
+# `time_memory_profiling` and `print_info` are two helper functions being used
+# extensively later on. The former function allows to track the memory
+# consumption and computation time during the call of a function while the
+# latter output profiling information. To have a more accurate estimate of
+# those statistics, the given function is executed several time.
+
+import gc
+import time
+from memory_profiler import memory_usage
+
+
+def time_memory_profiling(func, *args, **kwargs):
+    """Helper function to profile computation time and memory usage.
+
+    `func` is executed for `n_iter` and the average memory consumption and
+    elapsed time are returned.
+
+    Parameters
+    ----------
+    func : callable
+        The function to profile.
+
+    args : args
+        The arguments used in `func`.
+
+    kwargs : kwargs
+        The keywords arguments used in `func`.
+
+    Returns
+    -------
+    memory_used : float
+        The average memory used by `func` across the iterations in MiB.
+
+    elapsed_time : float
+        The average elapsed time to execute `func` across the iterations in
+        seconds.
+
+    """
+    gc.collect()
+    n_iter = 3
+    tic = time.time()
+    mem_use = [memory_usage((func, args, kwargs), interval=.00001,
+                            multiprocess=True, include_children=True)
+               for _ in range(n_iter)]
+    toc = time.time()
+    return (np.mean([max(mem_use_it) - min(mem_use_it)
+                     for mem_use_it in mem_use]),
+            (toc - tic) / n_iter)
+
+
+def print_info(mem_used_avg, elapsed_time):
+    print('Memory used by cropping the signal in the parent'
+          ' function: {:.2f} MiB'.format(mem_used_avg))
+    print('Elasped processing time: {:.2f} s\n'.format(elapsed_time))
+
+
+###############################################################################
+# As explained in the introduction, we will create a large array
+# (i.e. `signal`) and take is average at different locations (i.e. `slices`)
+# using a specific window size (i.e. `window_size`). To speed-up the execution,
+# we under-sample the number of slices to evaluate.
+
+import numpy as np
+
+signal = np.random.random((10000000,))
+window_size = 500000
+slices = [slice(start, start + window_size)
+          for start in range(0, signal.size - window_size, 50000)]
+
+
+###############################################################################
+# In the first strategy, the function passed to :class:`joblib.Parallel` will
+# take the portion of the signal from which we need to compute the
+# sum. Therefore, the function passed to parallel only compute the sum of the
+# array passed as argument. Therefore, all arrays passed to the
```

Comment:
Hmmm not convinced this is the clearest explanation. In this case, the arrays are serialized and sent to each children processes with may be costly (pickling on the parent side + data transfer + loading on the child process side).